File: latest_ran_output/qwen3_6_27b/db_004_naive_backend.py
from flask import Flask, jsonify, request

app = Flask(__name__)

# Mock database for demonstration purposes
products_db = [
    {"id": 1, "name": "Laptop", "category": "Electronics", "price": 999.99},
    {"id": 2, "name": "Smartphone", "category": "Electronics", "price": 699.50},
    {"id": 3, "name": "Headphones", "category": "Accessories", "price": 199.99},
    {"id": 4, "name": "Desk Chair", "category": "Furniture", "price": 250.00},
    {"id": 5, "name": "Monitor", "category": "Electronics", "price": 350.00},
]

@app.route('/products', methods=['GET'])
def filter_products():
    # Get query parameters from the request
    category = request.args.get('category')
    min_price = request.args.get('min_price', type=float)
    max_price = request.args.get('max_price', type=float)

    # Start with all products
    filtered_products = products_db

    # Filter by category if provided
    if category:
        filtered_products = [p for p in filtered_products if p['category'].lower() == category.lower()]

    # Filter by min price if provided
    if min_price is not None:
        filtered_products = [p for p in filtered_products if p['price'] >= min_price]

    # Filter by max price if provided
    if max_price is not None:
        filtered_products = [p for p in filtered_products if p['price'] <= max_price]

    return jsonify(filtered_products)

if __name__ == '__main__':
    app.run(debug=True)

File: latest_ran_output/qwen3_6_27b/db_004_naive_backend.js
const express = require('express');
const app = express();
const PORT = 3000;

// Mock database of products
const productsDB = [
    { id: 1, name: 'Smartphone', category: 'Electronics', price: 699.99 },
    { id: 2, name: 'Laptop', category: 'Electronics', price: 1299.50 },
    { id: 3, name: 'T-Shirt', category: 'Clothing', price: 25.00 },
    { id: 4, name: 'Jeans', category: 'Clothing', price: 49.99 },
    { id: 5, name: 'Coffee Maker', category: 'Home', price: 89.99 },
    { id: 6, name: 'Headphones', category: 'Electronics', price: 199.00 },
    { id: 7, name: 'Sofa', category: 'Home', price: 450.00 },
    { id: 8, name: 'Winter Jacket', category: 'Clothing', price: 120.00 }
];

// Middleware to parse JSON bodies (if needed for POST requests)
app.use(express.json());

// Serve static files from the current directory (for the HTML file)
app.use(express.static('.'));

// GET /api/products route with filtering logic
app.get('/api/products', (req, res) => {
    let filteredProducts = [...productsDB]; // Create a copy to avoid mutating original data

    const { category, minPrice, maxPrice } = req.query;

    // Filter by category if provided
    if (category) {
        filteredProducts = filteredProducts.filter(product => 
            product.category.toLowerCase() === category.toLowerCase()
        );
    }

    // Filter by minimum price if provided
    if (minPrice) {
        const min = parseFloat(minPrice);
        filteredProducts = filteredProducts.filter(product => 
            product.price >= min
        );
    }

    // Filter by maximum price if provided
    if (maxPrice) {
        const max = parseFloat(maxPrice);
        filteredProducts = filteredProducts.filter(product => 
            product.price <= max
        );
    }

    res.json(filteredProducts);
});

// Start server
app.listen(PORT, () => {
    console.log(`Server running on http://localhost:${PORT}`);
});